sqlglot/dialects/snowflake.py

```diff
@@ -167,14 +167,21 @@ def _datatype_sql(self: Snowflake.Generator, expression: exp.DataType) -> str:
     return self.datatype_sql(expression)
 
 
-def _regexpilike_sql(self: Snowflake.Generator, expression: exp.RegexpILike) -> str:
-    flag = expression.text("flag")
+def _regexplike_sql(self: Snowflake.Generator, expression: exp.RegexpILike | exp.RegexpLike) -> str:
+    flag = expression.args.get("flag")
 
-    if "i" not in flag:
-        flag += "i"
+    if isinstance(expression, exp.RegexpILike):
+        case_insensitive_flag = exp.Literal.string("i")
+        flag = self.func("CONCAT", case_insensitive_flag, flag) if flag else case_insensitive_flag
```

Comment:
I didn't add any tests for the `CONCAT` code path because I don't know of a dialect that actually supports `flags` as a third argument in `RegexpILike`. I can probably just add one anyway for the SQLGlot dialect if necessary.